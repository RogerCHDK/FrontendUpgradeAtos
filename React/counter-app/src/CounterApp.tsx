import React,{ useState} from 'react'

interface Props{
    value?: number
}

const CounterApp: React.FC<Props> = ( { value = 0 } ) =>{

    const [ counter , setCounter ] = useState( value );

    const handleAdd = () =>{
        setCounter( counter + 1); 
    }

    const handleReset = () =>{
        setCounter( value ); 
    }

    const handleRest = () =>{
        setCounter( counter - 1); 
    }

    return (
        <>
        <h1>CounterApp</h1>
        <h2>{ counter }</h2>
        <button onClick={ handleAdd } className='add'> +1 </button>
        <button onClick={ handleReset } className='reset'> Reset </button>
        <button onClick={ handleRest } className='substract'>  -1 </button>
        </>
    );
}

export default CounterApp;
