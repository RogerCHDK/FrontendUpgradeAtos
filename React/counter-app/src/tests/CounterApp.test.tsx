import { mount, shallow, ShallowWrapper } from "enzyme";
import CounterApp from '../CounterApp';

describe('Pruebas CounterApp', () => {

    let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach( () =>{
        wrapper = shallow( <CounterApp />);
    });
    
    test('should show the component CounteApp Succesfully', () => {
        
        // const wrapper = shallow( <CounterApp />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('should show the value by default', () => {
        const value: number = 10;
        const wrapper = shallow( <CounterApp value={value}/>);

        const valueByDefault: number = +wrapper.find('h2').text().trim();

        expect( valueByDefault ).toBe( value );
    });

    test('should increment in one when click on the button +1', () => {
        // const wrapper = shallow( <CounterApp />);
        const btnAdd = wrapper.find('.add');
        btnAdd.simulate('click');
        
        const newValue: number = +wrapper.find('h2').text().trim();

        expect( newValue ).toBe(1);
    });

    test('should substract -1 when click on the button', () => {
        const btnSubstract = wrapper.find('.substract');
        btnSubstract.simulate('click');
        
        const newValue: number = +wrapper.find('h2').text().trim();

        expect( newValue ).toBe(-1);
    });

    test('should reset the default value when click on the button', () => {
        // const value: number = 100;
        const wrapper = mount( <CounterApp value={100}/>);

        // Simular sumarle varias veces
        const btnAdd = wrapper.find('.add');
        btnAdd.simulate('click');
        btnAdd.simulate('click');
        btnAdd.simulate('click');

        // Resetear el valor
        const btnReset = wrapper.find('.reset');
        btnReset.simulate('click');

        // Obtener el valor del contador
        const newValue: number = +wrapper.find('h2').text().trim();
        // Obtener valor por defecto pasasdo por los props
        const valueProps: number = wrapper.props().value;

        expect( newValue ).toBe( valueProps );
    });
});