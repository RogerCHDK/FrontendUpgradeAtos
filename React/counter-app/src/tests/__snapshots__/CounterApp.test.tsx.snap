// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas CounterApp should show the component CounteApp Succesfully 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    0
  </h2>
  <button
    className="add"
    onClick={[Function]}
  >
     +1 
  </button>
  <button
    className="reset"
    onClick={[Function]}
  >
     Reset 
  </button>
  <button
    className="substract"
    onClick={[Function]}
  >
      -1 
  </button>
</Fragment>
`;
